assets/js/googlesitekit/datastore/user/authentication.test.js

```diff
@@ -0,0 +1,334 @@
+/**
+ * core/user Data store: Authentication info tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	createTestRegistry,
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/user authentication', () => {
+	const responseAuthenticated = { authenticated: true, requiredScopes: [], grantedScopes: [] };
+	const apiEndpoint = /^\/google-site-kit\/v1\/core\/user\/data\/authentication/;
```

Comment:
I'm not against this being shared between all tests but let's rename it to make the consuming code a bit clearer.
```suggestion
	const coreUserDataEndpointRegExp = /^\/google-site-kit\/v1\/core\/user\/data\/authentication/;
```